app/src/main/java/com/nutomic/syncthingandroid/views/WifiSsidPreference.java

```diff
@@ -59,99 +61,69 @@ public WifiSsidPreference(Context context) {
-            CharSequence[] all = extractSsid(networks, false);
-            filterRemovedNetworks(selected, all);
-            setEntries(extractSsid(networks, true)); // display without surrounding quotes
-            setEntryValues(all); // the value of the entry is the SSID "as is"
-            setValues(selected); // the currently selected values (without meanwhile deleted networks)
-            super.showDialog(state);
-        } else {
-            Toast.makeText(getContext(), R.string.sync_only_wifi_ssids_wifi_turn_on_wifi, Toast.LENGTH_LONG).show();
-        }
+        Context context = getContext();
+
+        Set<String> selected = getSharedPreferences().getStringSet(getKey(), new HashSet<>());
+        // from JavaDoc: Note that you must not modify the set instance returned by this call.
+        // therefore required to make a defensive copy of the elements
+        selected = new HashSet<>(selected);
+        List<String> all = new ArrayList<>(selected);
 
-        // On Android 8.1, ACCESS_COARSE_LOCATION is required, see issue #999
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O_MR1) {
-            if (ContextCompat.checkSelfPermission(getContext(), Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
-                if (getContext() instanceof Activity) {
-                    Activity activity = (Activity) getContext();
-                    ActivityCompat.requestPermissions(activity, new String[] { Manifest.permission.ACCESS_COARSE_LOCATION }, Constants.PERM_REQ_ACCESS_COARSE_LOCATION);
-                    this.getDialog().dismiss(); // wait for result
-                } else {
-                    Toast.makeText(getContext(), R.string.sync_only_wifi_ssids_need_to_grant_location_permission, Toast.LENGTH_LONG).show();
+        boolean connected = false;
+        WifiManager wifiManager = (WifiManager) context.getApplicationContext().getSystemService(Context.WIFI_SERVICE);
+        if (wifiManager != null) {
+            WifiInfo info = wifiManager.getConnectionInfo();
+            if (info != null) {
+                String ssid = info.getSSID();
+                // api lvl 30 will have WifiManager.UNKNOWN_SSID
+                if (ssid != "" && !ssid.contains("unknown ssid") && !selected.contains(ssid)) {
+                    all.add(ssid);
+                    connected = true;
                 }
             }
         }
-    }
 
-    /**
-     * Removes any network that is no longer saved on the device. Otherwise it will never be
-     * removed from the allowed set by MultiSelectListPreference.
-     */
-    private void filterRemovedNetworks(Set<String> selected, CharSequence[] all) {
-        HashSet<CharSequence> availableNetworks = new HashSet<>(Arrays.asList(all));
-        selected.retainAll(availableNetworks);
-    }
+        if (!connected) {
+            Toast.makeText(context, R.string.sync_only_wifi_ssids_connect_to_wifi, Toast.LENGTH_LONG).show();
```

Comment:
How do we know it's not a perms flop?
I guess we might not have asked for the permission at this point?